apis/Google.Cloud.AspNetCore.DataProtection.SecretManager/Google.Cloud.AspNetCore.DataProtection.SecretManager/CloudSecretManagerXmlRepository.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Api.Gax.Grpc;
+using Google.Api.Gax.ResourceNames;
+using Google.Cloud.SecretManager.V1;
+using Google.Protobuf;
+using Grpc.Core;
+using Microsoft.AspNetCore.DataProtection.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Xml.Linq;
+using Secret = Google.Cloud.SecretManager.V1.Secret;
+
+namespace Google.Cloud.AspNetCore.DataProtection.SecretManager;
+
+/// <summary>
+/// Implementation of <see cref="IXmlRepository"/> that stores the protected elements in a Google Cloud Secret Manager.
+/// This class is configured by <see cref="GoogleCloudDataProtectionBuilderExtensions.PersistKeysToGoogleCloudSecretManager(Microsoft.AspNetCore.DataProtection.IDataProtectionBuilder, string, string)"/>
+/// (and other overloads).
+/// </summary>
+internal class CloudSecretManagerXmlRepository : IXmlRepository
+{
+    private readonly SecretManagerServiceClient _secretManagerServiceClient;
+    private readonly SecretName _secretName;
+    private readonly ProjectName _projectName;
+    private readonly SecretVersionName _version;
+    private readonly CallSettings _callSettings = CallSettingsExtensions.WithRetry(CallSettings.FromExpiration(Expiration.FromTimeout(TimeSpan.FromMilliseconds(10000))), RetrySettings.FromExponentialBackoff(maxAttempts: 5, initialBackoff: TimeSpan.FromMilliseconds(0.2), maxBackoff: TimeSpan.FromMilliseconds(10000), backoffMultiplier: 1.5, retryFilter: RetrySettings.FilterForStatusCodes(StatusCode.DeadlineExceeded, StatusCode.Unavailable)));
```

Comment:
The default setting are just specifying the Expiration time but here I want to implement retry as well alongside. Its helps incase we are trying to read the value but the serivce was unavailable. Its better to retry for while instead of throwing an error rightaway (likewise in othercases) .
Thoughts?

Reply:
I think for `AccessSecretVersion` we already have retry in place out of the box for `Unavailable` and `ResourceExhausted` error codes. Do we need it for `AddSecretVersion` and `CreateSecret` ? Have we encountered any issues or are we doing it defensively for better resiliency? 

It may be fine and I 'll leave it for Jon/Amanda to take a call.

Reply:
Ok got it we can simply the `AccessSecretVersion` then for other I did it defensively only haven't encountered any issues.

Reply:
(So the question arises if retry could be configured for one API in the client, it could have been done for others as well but not done. So, there must be a reason for not retrying the Create and Add methods.) Let's take inputs from Jon/Amanda before making a change.

Reply:
I've removed the logic for retry. As other are non idempotency operations and it can lead to inconsistency. 
Happy to get feedback on the same.